3ds Max/Max2Babylon/Exporter/BabylonExporter.cs

```diff
@@ -88,8 +89,103 @@ public void CheckCancelled()
+                {
+                    IINode n = node.GetChildNode(i);
+                    IsMeshCollapsable(n,animationGroupList,ref collapsabeNodes);
+                }
+                return;
+            }
+
+            if (node.ActualINode.ObjectRef.GetMesh()==null)
+            {
+                for (int i = 0; i < node.NumChildren; i++)
+                {
+                    IINode n = node.GetChildNode(i);
+                    IsMeshCollapsable(n,animationGroupList,ref collapsabeNodes);
+                }
+                return;
+            }
+
+            if (node.IsSkinned())
+            {
+                string message = $"{node.Name} can't be flatten, because is skinned";
+                RaiseMessage(message, 0);
+                for (int i = 0; i < node.NumChildren; i++)
+                {
+                    IINode n = node.GetChildNode(i);
+                    IsMeshCollapsable(n,animationGroupList,ref collapsabeNodes);
+                }
+                return;
+
+            }
+            
+            if (node.IsInAnimationGroups(animationGroupList))
+            {
+                string message = $"{node.Name} can't be flatten, because is part of an AnimationGroup";
+                RaiseMessage(message, 0);
+                for (int i = 0; i < node.NumChildren; i++)
+                {
+                    IINode n = node.GetChildNode(i);
+                    IsMeshCollapsable(n,animationGroupList,ref collapsabeNodes);
+                }
+                return;
+            }
+
+            if (node.IsNodeTreeAnimated())
+            {
+                string message = $"{node.Name} can't be flatten, his hierachy contains animated node";
+                RaiseMessage(message, 0);
+                for (int i = 0; i < node.NumChildren; i++)
+                {
+                    IINode n = node.GetChildNode(i);
+                    IsMeshCollapsable(n,animationGroupList,ref collapsabeNodes);
+                }
+                return;
+            }
+
+            collapsabeNodes.Add(node);
+        }
+
+        public void FlattenHierarchy(IINode node)
+        {
+            IINode hierachyRoot = (node != null) ? node : Loader.Core.RootNode;
+            AnimationGroupList animationGroupList = new AnimationGroupList();
+            animationGroupList.LoadFromData();
+
+            List<IINode> collapsabeNodes = new List<IINode>();
+            IsMeshCollapsable(hierachyRoot, animationGroupList,ref collapsabeNodes);
+
+            foreach (IINode collapsabeINode in collapsabeNodes)
+            {
+                collapsabeINode.CollapseHierachy();
+            }
+        }
+
         public void Export(ExportParameters exportParameters)
         {
+            this.exportParameters = exportParameters;
+            IINode exportNode = (exportParameters as MaxExportParameters).exportNode;
+
+            if(exportParameters.flattenExport) FlattenHierarchy(exportNode);
```

Comment:
lets not inline if statements outside of ternary operators for assignment

Reply:
ok